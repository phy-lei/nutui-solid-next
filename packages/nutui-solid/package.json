{
  "name": "nutui-solid",
  "version": "1.0.0",
  "description": "",
  "author": "",
  "license": "ISC",
  "keywords": [],
  "main": "dist/nutui.solid.umd.js",
  "module": "dist/nutui.solid.es.js",
  "typings": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "prepare": "node scripts/generate-nutui.cjs",
    "generate:themes": "node scripts/generate-themes.cjs",
    "dev": "pnpm dev-components && pnpm build-css",
    "build": "pnpm build-components && pnpm build-css",
    "dev-components": "vite build --watch",
    "build-components": "vite build",
    "build-css": "pnpm generate:themes && vite build -c vite.config.build.css.ts"
  },
  "dependencies": {
    "classnames": "^2.5.1"
  },
  "devDependencies": {
    "@babel/core": "^7.23.3",
    "@rollup/plugin-commonjs": "^25.0.7",
    "@rollup/plugin-node-resolve": "^15.2.3",
    "@types/babel__core": "^7.20.4",
    "@types/jest": "^29.5.11",
    "autoprefixer": "^10.4.19",
    "babel-loader": "^9.1.3",
    "babel-plugin-transform-solid-jsx": "4.0.0-beta.85",
    "fs-extra": "^11.2.0",
    "jest": "^29.7.0",
    "merge-anything": "^5.1.7",
    "postcss-import": "^16.1.0",
    "rollup": "^4.9.5",
    "rollup-plugin-node-externals": "^4.0.0",
    "rollup-plugin-ts": "^3.4.5",
    "sass": "^1.77.5",
    "solid-refresh": "^0.6.3",
    "vite": "^5.1.3",
    "vite-plugin-dts": "3.6.4",
    "vitefu": "^0.2.5"
  }
}
