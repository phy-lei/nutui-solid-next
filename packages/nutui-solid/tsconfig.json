{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "jsx": "preserve",
    "jsxImportSource": "solid-js",
    "types": ["vite-plugin-pwa/info"],
    "paths": {
      "@/*": ["src/*"],
      "nutui-solid": ["./dist"]
    },
    "resolveJsonModule": true,
    "esModuleInterop": true
  },
  "include": ["**/*.ts", "**/*.tsx"]
}
