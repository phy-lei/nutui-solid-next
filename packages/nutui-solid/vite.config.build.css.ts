import { defineConfig } from 'vite'

export default defineConfig({
  build: {
    minify: true,
    lib: {
      entry: './dist/styles/themes/default.scss',
      formats: ['es'],
      name: 'style',
      fileName: 'style'
    },
    emptyOutDir: false
  }
})
