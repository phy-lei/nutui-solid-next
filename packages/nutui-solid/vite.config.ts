import { defineConfig } from 'vite'
import dts from 'vite-plugin-dts'
import fse from 'fs-extra'
import path from 'path'
import config from './package.json'
import solidPlugin from './plugin/vite-plugin-solid'

const banner = `/*!
* ${config.name} v${config.version} ${new Date()}
* (c) 2023 @jdf2e.
* Released under the MIT License.
*/`

const { resolve } = path

let fileStr = `@import "@/styles/variables.scss";`
const projectID = process.env.VITE_APP_PROJECT_ID
if (projectID) {
  fileStr = `@import '@/styles/variables-${projectID}.scss';`
}
// https://vitejs.dev/config/
export default defineConfig({
  resolve: {
    alias: [{ find: '@', replacement: resolve(__dirname, './src') }],
  },
  plugins: [
    solidPlugin(),
    dts({
      outDir: 'dist/types',
      clearPureImport: false,
      include: [
        "src/components/nutui.solid.build.ts",
        "src/components/**/*.ts",
        "src/components/**/*.tsx",
      ],
      exclude: [
        'node_modules/**',
        'src/components/**/demos/**',
        'src/**/*.spec.tsx',
        'plugin/**'
      ],
      beforeWriteFile(filePath, content) {
        return { filePath: filePath.replace('src/', ''), content }
      },
      afterBuild: () => {
        fse
          .readFile('./dist/types/nutui.solid.build.d.ts', 'utf-8')
          .then((data: string) => {
            fse.remove('./dist/types/nutui.solid.build.d.ts')
            fse.outputFile(
              './dist/types/index.d.ts',
              `${data.replace(/\.\.\//g, './')}`
            )
          })
      },
    }),
  ],
  build: {
    minify: false,
    emptyOutDir: true,
    rollupOptions: {
      // 请确保外部化那些你的库中不需要的依赖
      external: ['solid-js'],
      output: [
        {
          banner,
          globals: {
            solidJs: 'solidJs',
          },
          format: 'es',
          entryFileNames: 'nutui.solid.es.js',
        },
        {
          banner,
          globals: {
            solidJs: 'solidJs',
          },
          name: 'nutui.solid',
          format: 'umd',
          entryFileNames: 'nutui.solid.umd.js',
        },
      ],
    },
    lib: {
      entry: 'src/components/nutui.solid.build.ts',
    },
  },
})
