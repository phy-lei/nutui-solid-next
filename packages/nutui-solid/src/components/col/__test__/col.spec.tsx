import { test, expect } from "vitest"
import { render } from "@solidjs/testing-library"
import { Row, Col } from 'nutui-solid'

test('should render Col correctly and should render gutter correctly', async () => {
  const { container } = render(() => <Row gutter="6">
  <Col span="8">8</Col>
  <Col span="8">8</Col>
  <Col span="8">8</Col>
</Row>)
expect(container.children[0]).toMatchSnapshot()
})
