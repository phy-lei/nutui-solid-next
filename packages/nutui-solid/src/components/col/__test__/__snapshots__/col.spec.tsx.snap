// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Col correctly and should render gutter correctly 1`] = `
<div
  class="nut-row nut-row- nut-row-justify-start nut-row-align-flex-start nut-row-flex-nowrap"
>
  <div
    class="nut-col nut-col-gutter nut-col-8 nut-col-offset-0"
    style="padding-left: 3px; padding-right: 3px;"
  >
    8
  </div>
  <div
    class="nut-col nut-col-gutter nut-col-8 nut-col-offset-0"
    style="padding-left: 3px; padding-right: 3px;"
  >
    8
  </div>
  <div
    class="nut-col nut-col-gutter nut-col-8 nut-col-offset-0"
    style="padding-left: 3px; padding-right: 3px;"
  >
    8
  </div>
</div>
`;
